delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/indexing/CompositeViewsCoordinator.scala

```diff
@@ -36,42 +30,30 @@ object CompositeViewsCoordinator {
     *   a cache of the current running views
     * @param supervisor
     *   the general supervisor
-    * @param compilePipeChain
-    *   how to compile pipe chains
-    * @param graphStream
-    *   to provide the data feeding the indexing
-    * @param buildSpaces
-    *   to create the necessary namespaces / indices and the pipes/sinks to interact with them
-    * @param compositeProjections
-    *   to fetch/save progress as well as handling restarts
+    * @param lifecycle
+    *   how to pouet
     */
   final private class Active(
       fetchViews: Offset => ElemStream[CompositeViewDef],
       cache: KeyValueStore[ViewRef, ActiveViewDef],
       supervisor: Supervisor,
-      compilePipeChain: PipeChain.Compile,
-      graphStream: CompositeGraphStream,
-      buildSpaces: ActiveViewDef => CompositeSpaces,
-      compositeProjections: CompositeProjections
-  )(implicit cr: RemoteContextResolution)
-      extends CompositeViewsCoordinator {
+      lifecycle: CompositeProjectionLifeCycle
+  ) extends CompositeViewsCoordinator {
 
     def run(offset: Offset): Stream[Task, Elem[Unit]] = {
```

Comment:
I like the result result on the `run` function, it's simpler now

Reply:
I think it was needed yes as the coordinator knew too much.
And this change should also be helpful to implement #3948